src/middlewares/authenticator.js

```diff
@@ -0,0 +1,111 @@
+import jwt from 'jsonwebtoken';
+import bcrypt from 'bcryptjs';
+import 'dotenv/config';
+
+/**
+ * @class AuthenticateUser
+ * @description Authenticates a given user
+ * @exports AuthenticateUser
+ */
+class Authenticator {
+  /**
+   * @static
+   * @param {any} password - The user password to be hashed
+   * @returns {object} - JSON response object
+   *
+   * @memberOf Authenticator
+   */
+  static hashPassword(password) {
+    return bcrypt.hashSync(password, bcrypt.genSaltSync(10));
+  }
+
+  /**
+   * @static
+   * @param {any} password The user password
+   * @param {any} hashPassword The hashed password
+   * @returns {object} - JSON response object
+   *
+   * @memberOf Authenticator
+   */
+  static comparePassword(password, hashPassword) {
+    return bcrypt.compareSync(password, hashPassword);
+  }
+
+  /**
+   * @static
+   * @param {any} user The user object
+   * @returns {object} - JSON response object
+   *
+   * @memberOf Authenticator
+   */
+  static generateToken(user) {
+    const token = jwt.sign({ user }, process.env.JWTKEY, { expiresIn: '24h' });
+    return token;
+  }
+
+  /**
+   * @method isAuthenticated
+   * @description Verifies that user is authenticated
+   * @param {object} req - The Request Object
+   * @param {object} res - The Response Object
+   * @param {object} next - The next function
+   * @returns {object} - JSON response object
+   */
+  static isAuthenticated(req, res, next) {
```

Comment:
Can we employ a better name for our method?

Reply:
Adjusted as recommended, thanks